Sources/protoc-gen-swift/FieldGenerator.swift

```diff
@@ -110,6 +113,50 @@ class FieldGeneratorBase {
         }
     }
 
+    func writeProtoNameInstruction(to writer: inout ProtoNameInstructionWriter) {
+        // Protobuf Text uses the unqualified group name for the field
+        // name instead of the field name provided by protoc.  As far
+        // as I can tell, no one uses the fieldname provided by protoc,
+        // so let's just put the field name that Protobuf Text
+        // actually uses here.
+        let protoName: String
+        if fieldDescriptor.internal_isGroupLike {
+            protoName = fieldDescriptor.messageType!.name
+        } else {
+            protoName = fieldDescriptor.name
+        }
+        let jsonName = fieldDescriptor.jsonName
+
+        // TODO: Create a separate instruction for the group being able to match
```

Comment:
Seems like we should do this extra opcode now vs. keeping it as a TODO

Reply:
Added a new opcode for groups.